editNickname.php

```diff
@@ -0,0 +1,18 @@
+<?php
+require_once "functions/allFunctions.php";
+
+if (!isset($_POST['id']) || !is_numeric($_POST['id'])){
+    header("location: index.php");
+    exit();
+}
+
+$db = connectToDB('pokemon-collection');
+$inputtedNewNickname = $_POST['newNickname'];
+$inputtedID = $_POST['id'];
+
+$cleanNickname = sanitiseText($inputtedNewNickname);
```

Comment:
Worth checking the length of this to make sure it's not too long, this check plus the HTML maxlength attribute should make this pretty bullet proof

Reply:
You could also pass $_POST['newNickname'] directly though your sanitiser to save a variable